### PR TITLE
feat: add open exposed url to pod details

### LINKED ISSUE #1349: Open Browser actions should handle multiple ports
### Is your enhancement related to a problem? Please describe

If I open a single port when starting a container, I can quickly open the exposed port of my container.

But if I open more than one port, then I have no way to open a selected port, it's taking the first one.

### Describe the solution you'd like

For Kebab menu, it could be one entry per port like 'Open 8080 port in the Browser' or  a drop-down after clicking on 'Open Browser' it could display sub-entries 'Open 8080, Open 9000', etc.

For icon in the container details it could bring a drop-down if there is more than one port

For Ports displayed in the container list, we should be able to click on the ports
image

### Describe alternatives you've considered

any proposal to be able to select a given port

### Additional context

_No response_

### LINKED ISSUE #2978: Ability to open application in the browser for Pod
### Is your feature request related to a problem? Please describe

When running an application in a pod, I may have a container that runs an application which could be reached by an exposed port. 

Right now, as a user, I need to go to the summary of the pod, click on the container and then I can open the app in a browser (from the top right button).

### Describe the solution you'd like

As a user, it would be nice if it could be possible to do that directly from the pod's details view. An action button which could eventually propose to the user which exposed port to open in a browser. 

### Describe alternatives you've considered

Enhancing the summary view of the pod's details page. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vzhukovs**
### What does this PR do?

Adds a button to pod details to open the exposed url.

### What issues does this PR fix or reference?

#2978 

### How to test this PR?

There is an open url button near refresh button in top right corner of pod details.

### CODE REVIEW

File: packages/renderer/src/lib/pod/PodActions.svelte [hunk 1]
```diff
@@ -119,6 +142,18 @@ if (dropdownMenu) {
       menu="{dropdownMenu}"
       detailed="{detailed}"
       icon="{faRocket}" />
+    {#if openingUrls !== undefined}
+      {#each openingUrls as url}
+        <ListItemButtonIcon
+          title="Open Browser"
+          onClick="{() => window.openExternal(url)}"
```
Comment (benoitf) [inline packages/renderer/src/lib/pod/PodActions.svelte]: from my understanding user will see "Open Browser" several times if I have like 4 URLs
but how to differentiate each link if I only see "Open Browser"

also does it mean we will have like 5 icons if I have 5 links ?

File: packages/renderer/src/lib/pod/PodDetails.spec.ts [hunk 2]
```diff
@@ -51,6 +52,7 @@ const removePodMock = vi.fn();
 
 beforeAll(() => {
   (window as any).listPods = listPodsMock;
+  (window as any).listContainers = listContainersMock.mockResolvedValue([]);
   (window as any).kubernetesListPods = kubernetesListPodsMock;
   (window as any).removePod = removePodMock;
 });
```
Comment (benoitf) [inline packages/renderer/src/lib/pod/PodDetails.spec.ts]: we're missing the test of checking if we have links to open URLs and verify that if we click on the button we're redirected to the expected link

**Comment by jeffmaury** [comment]
Tried on a container without port (node) and can see the Open browser link on the infra node

**Review by jeffmaury — COMMENTED** [review]
See previous comment

**Comment by vzhukovs** [comment]
@benoitf @jeffmaury can you check it, please. I've provided some updates to the current PR:

Terminated Pod:


Active Pod:


Opening Dialogs:



Single Port:

**Comment by deboer-tim** [comment]
The images look good, but could we make the button disabled when the pod is stopped instead of hidden? Otherwise there is some visual movement and the start/stop button changes location.

**Comment by benoitf** [comment]
maybe a UX question, do we need to display a dropdown/single action when we have only one item rather than just clicking on the button ?

**Comment by deboer-tim** [comment]
> maybe a UX question, do we need to display a dropdown/single action when we have only one item rather than just clicking on the button ?

+1, I would prefer if there's only one there's no drop-down and we just open it. (or maybe longer term there is a separate drop down to select which port and the button always opens?)

**Comment by vzhukovs** [comment]
> The images look good, but could we make the button disabled when the pod is stopped instead of hidden? Otherwise there is some visual movement and the start/stop button changes location.

> maybe a UX question, do we need to display a dropdown/single action when we have only one item rather than just clicking on the button ?

There isn't a problem to make it disable and actionable when we have only one exposed ports. Going to update the PR.

**Comment by vzhukovs** [comment]
@deboer-tim @benoitf I've updated the PR, now the situation with buttons following:

If pod doesn't have exposed port, button will be visible but disabled, the same state either on running or stopped pod:


If pod has only one exposed port, button will handle openning port if pod started, otherwise button visible, but disabled.


If pod has several exposed ports, then on stopped state button will display dropdown with disabled buttons, the opposite situation when pod is started, in this case buttons vill be enabled and handle opening particular port.

**Comment by benoitf** [comment]
I tried and it's displaying several times the ports
image

I created a container using
```
podman run -p 8080:80 -p 8085:80 -p 8086:70 nginx
```

then convert this container to a pod

and then opening the details of the pod

**Review by benoitf — CHANGES_REQUESTED** [review]
on my side it's not working, I saw several duplicated items

**Comment by vzhukovs** [comment]
> on my side it's not working, I saw several duplicated items

going to check it

**Review by benoitf — APPROVED** [review]

**Pull request merged**